phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -602,7 +616,7 @@ private Collection<? extends Mutation> groupMutations(MiniBatchOperationInProgre
             // unfortunately, we really should ask if the raw mutation (rather than the combined mutation)
             // should be indexed, which means we need to expose another method on the builder. Such is the
             // way optimization go though.
-            if (miniBatchOp.getOperationStatus(i) != IGNORE && this.builder.isEnabled(m)) {
+            if (miniBatchOp.getOperationStatus(i).getResult() == null && this.builder.isEnabled(m)) {
```

Comment:
@jinggou Do we need this change ? Will the original change not work

Reply:
@tkhurana Yes, the original code doesn't work because we changed the code `miniBatchOp.setOperationStatus(i, IGNORE);` as @virajjasani mentioned below.